torchmetrics/collections.py

```diff
@@ -172,14 +185,16 @@ def persistent(self, mode: bool = True) -> None:
         for _, m in self.items():
             m.persistent(mode)
 
-    def _set_prefix(self, k: str) -> str:
-        return k if self.prefix is None else self.prefix + k
+    def _set_name(self, k: str) -> str:
+        out = k if self.prefix is None else self.prefix + k
+        out = out if self.postfix is None else out + self.postfix
+        return out
```

Comment:
```suggestion
    def _set_name(self, base: str) -> str:
        name = base if self.prefix is None else self.prefix + base
        name = name if self.postfix is None else name + self.postfix
        return name
```
the k is no meaning...